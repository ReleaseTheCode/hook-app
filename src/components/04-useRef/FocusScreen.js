import React, { useRef } from 'react'

export const FocusScreen = () => {

    const inputRef = useRef();

    const handleClick = () => {
        inputRef.current.select();
    }
    return (
        <div className='m-3'>
            <h1>Focus Screen</h1>
            <hr/>
            <input
                ref={ inputRef }
                className='form-control mb-2'
                placeholder='Su nombre'
            />
            <button
                className='btn btn-secondary'
                onClick={ handleClick }>
                Focus
            </button>
        </div>
    )
}
