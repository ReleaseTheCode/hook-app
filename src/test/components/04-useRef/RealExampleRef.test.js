import React from 'react';
import { shallow } from 'enzyme';
import { RealExampleRef } from "../../../components/04-useRef/RealExampleRef"

describe('Test on RealExampleRef', () => {
    const wrapper = shallow(<RealExampleRef/>);

    test('should render', () => {
        expect( wrapper ).toMatchSnapshot()
    });

    test('should change state clicking button', () => {
        wrapper.find('button.btn').simulate('click');
        expect( wrapper.find('MultipleCustomHooks').exists() ).toBe(true)
    })
    
    
})
