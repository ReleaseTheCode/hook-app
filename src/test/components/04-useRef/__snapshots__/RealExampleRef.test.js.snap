// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on RealExampleRef should render 1`] = `
<div>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-secondary"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
